### PR TITLE
New Script GetInstances

### PR DESCRIPTION
**Pull request by esharf**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
contributor request.

## Description
the script returns the instances name, brand, state, don't use by default, and category.
also possible filter by the brand (except also an array of brands) and by state (active/ disabled).

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/CommonScripts/Scripts/GetInstances/GetInstances.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import demistomock as demisto
+from CommonServerPython import *
+
+INTERNAL_MODULES_BRANDS = ['Scripts', 'Builtin', 'testmodule']
+ENABLED = 'active'
+DISABLED = 'disabled'
+
+
+def prepare_filter_list(args: Dict) -> List[Callable[[Dict], bool]]:
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/GetInstances/GetInstances.py:11]: this solution is nice but it is more complex than it should, please simplify it.

File: Packs/CommonScripts/Scripts/GetInstances/GetInstances.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+args:
+- name: brand
+  description: Filter by the brand name
+  isArray: true
+- name: is_enabled
+  auto: PREDEFINED
+  predefined:
+  - "true"
+  - "false"
+  description: Filter by the instance status
```
Comment (bakatzir) [inline Packs/CommonScripts/Scripts/GetInstances/GetInstances.yml:19]: let's add a defaultvalue of `true`

File: Packs/CommonScripts/ReleaseNotes/1_2_85.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### New: GetInstances
+- Returns all the Instance that existing in xsoar
```
Comment (bakatzir) [inline Packs/CommonScripts/ReleaseNotes/1_2_85.md:4]:
```suggestion
- Returns all the instances existing in Cortex XSOAR.
```

**Review by bakatzir — COMMENTED** [review]
How will this script work in a MT env? please check and let's revise if it will require changes in the code.

File: Packs/ModulesManagement/Scripts/GetInstances/GetInstances.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+INTERNAL_MODULES_BRANDS = ['Scripts', 'Builtin', 'testmodule']
+
+
+def filter_config(module: Dict, filter_brand: Optional[List[str]] = None, is_enabled: str = 'active'):
+    brand = module.get('brand')
+    if brand in INTERNAL_MODULES_BRANDS:
+        return False
+    elif filter_brand and brand not in filter_brand:
+        return False
+    elif not is_enabled == 'both' and not module.get('state') == is_enabled:
```
Comment (yaakovi) [inline Packs/ModulesManagement/Scripts/GetInstances/GetInstances.py:15]:
```suggestion
    elif is_enabled != 'both' and module.get('state') != is_enabled:
```

File: Packs/ModulesManagement/Scripts/GetInstances/GetInstances.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            yield config
+
+
+def main():
+    try:
+        args = prepare_args(demisto.args())
+        context_config = list(filter_instances(demisto.getModules(), **args))
+        return_results(CommandResults(
+            outputs=context_config,
+            outputs_prefix='Modules'
```
Comment (yaakovi) [inline Packs/ModulesManagement/Scripts/GetInstances/GetInstances.py:44]:
```suggestion
            outputs_prefix='Modules',
```

File: Packs/ModulesManagement/pack_metadata.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{
+    "name": "ModulesManagement",
+    "description": "Manage your instances on demisto",
```
Comment (yaakovi) [inline Packs/ModulesManagement/pack_metadata.json:3]: @esharf @ShirleyDenkberg please improve this description

**Review by yaakovi — APPROVED** [review]

**Comment by kirbles19** [comment]
@esharf Done.
if it isn't too late, consider changing the argument `is_enabled` to `status` or `instance_status`

**Pull request merged**